### PR TITLE
Resource-optimized placement

### PR DESCRIPTION
**Pull request by ledjon-behluli**
Adds resource-optimized placement to the grain placement page

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/orleans/grains/grain-placement.md](https://github.com/dotnet/docs/blob/01023e88a774095cd40f7633dd10f65aecf1449a/docs/orleans/grains/grain-placement.md) | [Grain placement](https://review.learn.microsoft.com/en-us/dotnet/orleans/grains/grain-placement?branch=pr-en-us-39745) |

### CODE REVIEW

File: docs/orleans/grains/grain-placement.md [hunk 1]
```diff
@@ -44,6 +44,18 @@ This placement strategy is configured by adding the <xref:Orleans.Concurrency.St
 
 A deterministic placement strategy that places grains on silos with a specific role. This placement strategy is configured by adding the <xref:Orleans.Placement.SiloRoleBasedPlacementAttribute> to a grain.
 
+## Resource-optimized placement
+
+This placement strategy attempts to optimize cluster resources by balancing grain activations across silos based on available memory and CPU usage. It assigns weights to runtime statistics to prioritize different resources and calculates a normalized score for each silo.  The silo with the lowest score is chosen for placing the upcoming activation. Normalization ensures that each property contributes proportionally to the overall score. Weights can be adjusted via the <xref:Orleans.Configuration.ResourceOptimizedPlacementOptions> based on user-specific requirements and priorities for different resources.
```
Comment (IEvangelist) [inline docs/orleans/grains/grain-placement.md:49]:
```suggestion
The resource-optimized placement strategy attempts to optimize cluster resources by balancing grain activations across silos based on available memory and CPU usage. It assigns weights to runtime statistics to prioritize different resources and calculates a normalized score for each silo. The silo with the lowest score is chosen for placing the upcoming activation. Normalization ensures that each property contributes proportionally to the overall score. Weights can be adjusted via the <xref:Orleans.Configuration.ResourceOptimizedPlacementOptions> based on user-specific requirements and priorities for different resources.
```

File: docs/orleans/grains/grain-placement.md [hunk 2]
```diff
@@ -44,6 +44,18 @@ This placement strategy is configured by adding the <xref:Orleans.Concurrency.St
...
+
+In addition, this placement strategy exposes an option to build a stronger *preference* to the local silo (*the one which got the request for making a new placement*) to be picked as the target for the activation. This is controlled via the `LocalSiloPreferenceMargin` property which is part of the options.
```
Comment (IEvangelist) [inline docs/orleans/grains/grain-placement.md:51]:
```suggestion
Additionally, this placement strategy exposes an option to build a stronger *preference* to the local silo (*the one which got the request for making a new placement*) to be picked as the target for the activation. This is controlled via the `LocalSiloPreferenceMargin` property which is part of the options.
```

**Review by IEvangelist — APPROVED** [review]

**Comment by ledjon-behluli** [comment]
@IEvangelist 8.1-preview

**Comment by BillWagner** [comment]
closing and reopening for a fresh CI build.

**Pull request merged**